components/dashboards-web-component/src/designer/components/WidgetConfigurationPanel.jsx

```diff
@@ -106,15 +223,196 @@ class WidgetConfigurationPanel extends React.Component {
         return publishers;
     }
 
+
+    getPreferences() {
+
+        /**
+         * helper function to get the page from a given dashboard given the pageId
+         * */
+        function getPage(dashboard, pageID) {
+            let pages = dashboard.pages;
+            if (pages) {
+                let pageResult = _.filter(pages, function (page) {
+                    return page.id === pageID;
+                })
+                return pageResult[0];
+            }
+            return null;
+        }
+
+        /**
+         * helper function to get the widget from a given page given the widgetId
+         * */
+        function searchForWidget(page, id) {
```

Comment:
Please move this function out of the function scope of the getPreferences()

Reply:
fixed in 87b573730d4c8021f11a7fc89783b583eed6edbf

Reply:
fixed with 87b573730d4c8021f11a7fc89783b583eed6edbf